### DISCUSSION TITLE
Feature Request: Configurable Rate Limiting for API Prompts

### DISCUSSION THREAD

**Discussion by Steve-Sexton** [cline/cline#1017] [Feature Requests]
**Problem**: Need to prevent API rate limit issues and manage costs by controlling prompt frequency

**Proposed Solution**: 
- User-configurable rate limiting
- Two modes: Time-based delay or requests per minute
- CLI flag to enable/disable
- Configurable parameters for timing/count

**Implementation Details**:
- Time mode: Fixed delay between requests
- Count mode: Max requests per time window
- Settings via CLI flags or config file
- Real-time metrics display (optional)

**Benefits**:
1. Prevents API throttling
2. Controls costs
3. Predictable performance
4. User flexibility

**Comment by kangax** [comment]
This is especially relevant now that free gem 2.5 tier is [severely limited](https://x.com/officiallogank/status/1922357621178200248?s=46&t=B57FHiiKxIOUOZ8y2NUchQ). Imagine if you could start a task that slowly queries model once a minute, then go have lunch and let it do its thing, slowly making progress over an hour.

**Comment by lightfinesthour** [comment]
openrouter also has a daily limit for free use, an api rate limit would be beneficial for anyone on a budget!

**Comment by lightfinesthour** [comment]
FYI : The cline branch called "Roo" already has this feature built in, I have switched to that branch to have this feature.
You can set a Rate limit in seconds after each model.
